lms/djangoapps/instructor/views/api.py

```diff
@@ -3663,7 +3663,8 @@ def _ban_student_from_course(course_id, usernames_expiry_key_value):
         if not check_expiry_validity(raw_user, expiry, invalid_usernames):
             continue
 
-        if re.search(r'^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w+$', raw_user):
+        raw_email = re.findall(r'[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+', raw_user)
+        if raw_email and raw_email[0] == raw_user:
```

Comment:
this can break when there is no matched record.

Reply:
I have used `raw_email and` it will not go further because of `and` so it will not break and yes tested it on all valid and invalid emails